satpy/readers/iasi_ng_l2_nc.py

```diff
@@ -0,0 +1,418 @@
+            full_group_name = base_name + group_name
+            self.file_content[full_group_name] = group_obj
+            self._collect_attrs(full_group_name, group_obj)
+            self.collect_metadata(full_group_name, group_obj)
+            self.collect_dimensions(full_group_name, group_obj)
+
+    def available_datasets(self, configured_datasets=None):
+        """Determine automatically the datasets provided by this file.
+
+        Uses a per product type dataset registration mechanism.
+        """
+        # pass along existing datasets
+        for is_avail, ds_info in configured_datasets or []:
+            yield is_avail, ds_info
+
+        for _, ds_info in self.dataset_infos.items():
+            yield True, ds_info
+
+    def register_dataset(self, ds_name, desc):
+        """Register a simple dataset given its name and a desc dict."""
+        if ds_name in self.dataset_infos:
+            raise KeyError(f"Dataset for {ds_name} already registered.")
+
+        ds_infos = {
+            "name": ds_name,
+            "sensor": "iasi_ng",
+            "file_type": self.filetype_info["file_type"],
+        }
+
+        ds_infos.update(desc)
+
+        self.dataset_infos[ds_name] = ds_infos
+
+    def process_dimension(self, key, value):
+        """Process a dimension entry from the file_content."""
+        dim_name = key.split("/")[-1]
+
+        if dim_name in self.dimensions_desc and self.dimensions_desc[dim_name] != value:
+            # This might happen if we have the same dim name from different groups:
+            raise KeyError(f"Detected duplicated dim name: {dim_name}")
+
+        self.dimensions_desc[dim_name] = value
+
+    def process_attribute(self, key, value):
+        """Process a attribute entry from the file_content."""
+        var_path, aname = key.split("/attr/")
+
+        if var_path not in self.variable_desc:
+            # maybe this variable is ignored, or this is a group attr.
+            return
+
+        self.variable_desc[var_path]["attribs"][aname] = value
+
+    def process_variable(self, key):
+        """Process a variable entry from the file_content."""
+        shape = self.file_content[f"{key}/shape"]
+
+        if np.prod(shape) <= 1:
+            # Ignoring scalar variable.
+            return
+
+        # Check if this variable should be ignored:
+        if any(p.search(key) is not None for p in self.ignored_patterns):
+            # Ignoring variable on user request.
+            return
+
+        # Prepare a description for this variable:
+        prefix, var_name = key.rsplit("/", 1)
+        dims = self.file_content[f"{key}/dimensions"]
+        dtype = self.file_content[f"{key}/dtype"]
+
+        desc = {
+            "location": key,
+            "prefix": prefix,
+            "var_name": var_name,
+            "shape": shape,
+            "dtype": f"{dtype}",
+            "dims": dims,
+            "attribs": {},
+        }
```

Comment:
that comment here surely means this code bit should be in its own function called eg `prepare_description_for_variable`?

Reply:
Hi @mraspaud ,

Thank you very much for this initial feedback 😊

I've read the article you mentioned, and even if I don't fully agree to this perspective on code comments I think I see your point, and will start working on the requested changes asap. I'll let you when all the fixes are integrated and you can have another look at this PR.

Thanks 🙏!

Reply:
Hello @mraspaud ,

I have now updated the content of the iasi_ng_l2_nc.py file removing those excessive comments (just keeping a couple of "dev notes" in the end). 

Now, just to be sure before continuing: you would also expect me to do the same for the test_iasi_ng_l2_nc.py file, right :-)?

Also, I'm not sure if you would expect me to "resolve" each of the points reported above myself, or if you prefer to review the changes first and then resolve those points yourself when you juge the result acceptable (please just let me know if you think I should do it myself).

Thanks!

Reply:
Thanks a lot, looks much better now!
yes it would be great if you could do the same in the test file.
Regarding the resolution of comments, I like it when the original commenter (in this case me) clicks the resolve button.

Reply:
Okay 👌copy that: so I'm updating the test file now.

Reply:
Hi @mraspaud ,

I have now also updated the test_iasi_ng_l2_nc.py file as discussed above. So please let me know when you find some time to give this PR another look, if you notice anything else that we should discuss/change. 

Thanks in advance for your time on this 🙏!